src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgeryTokenSerializer.cs

```diff
@@ -156,7 +161,14 @@ public string Serialize(AntiforgeryToken token)
 
                 writer.Flush();
                 var stream = serializationContext.Stream;
-                return WebEncoders.Base64UrlEncode(_cryptoSystem.Protect(stream.ToArray()));
+                var bytes = _cryptoSystem.Protect(stream.ToArray());
+
+                var count = bytes.Length;
+                var charsRequired = WebEncoders.GetArraySizeRequiredToEncode(count);
+                var chars = serializationContext.GetChars(charsRequired);
+                var outputLength = WebEncoders.Base64UrlEncode(bytes, 0, count, chars, 0);
+
+                return new string(chars, 0, outputLength);
```

Comment:
Can't avoid this allocation w/o changing the properties in `AntiforgeryTokenSet` from `string` to `ArraySegment<char>` or similar. Then would need `serializationContext.GetChars1()` and `serializationContext.GetChars2()` for the two tokens. Ick all 'round.